src/praisonai-agents/praisonaiagents/managed/shared_compute.py

```
@@ -0,0 +1,225 @@
+                agent.tools = merged
+                agent.backstory = (
+                    f"{original_backstory}\n\n{CAPABILITY_PROMPT}"
+                    if original_backstory
+                    else CAPABILITY_PROMPT
+                )
+                # The system prompt is memoised per (role, goal, tools); drop the
+                # cache so the capability text is actually rendered.
+                cache = getattr(agent, "_system_prompt_cache", None)
+                if cache is not None and hasattr(cache, "clear"):
+                    cache.clear()
+            except Exception as exc:  # pragma: no cover - never break a run over this
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)